src/robusta/core/sinks/pushover/pushover_sink.py

```diff
@@ -0,0 +1,119 @@
+from tabulate import tabulate
+
+from robusta.core.reporting.base import BaseBlock, Finding, FindingSeverity, FindingStatus
+from robusta.core.reporting.blocks import FileBlock, MarkdownBlock, TableBlock
+from robusta.core.sinks.sink_base import SinkBase
+from robusta.core.sinks.pushover.pushover_client import PushoverClient
+from robusta.core.sinks.pushover.pushover_sink_params import PushoverSinkConfigWrapper
+from robusta.core.sinks.transformer import Transformer
+from robusta.core.reporting.utils import is_image 
+
+
+SEVERITY_EMOJI_MAP = {
+    FindingSeverity.INFO: "\U0001F7E2",
+    FindingSeverity.LOW: "\U0001F7E1",
+    FindingSeverity.MEDIUM: "\U0001F7E0",
+    FindingSeverity.HIGH: "\U0001F534",
+}
+INVESTIGATE_ICON = "\U0001F50E"
+SILENCE_ICON = "\U0001F515"
+VIDEO_ICON = "\U0001F3AC"
+
+
+class PushoverSink(SinkBase):
+    def __init__(self, sink_config: PushoverSinkConfigWrapper, registry):
+        super().__init__(sink_config.pushover_sink, registry)
+
+        self.client = PushoverClient(sink_config.pushover_sink.token,
+                                     sink_config.pushover_sink.user,
+                                     sink_config.pushover_sink.device)
+        
+        self.send_files = sink_config.pushover_sink.send_files
+        self.send_as_html = sink_config.pushover_sink.send_as_html
+
+    def write_finding(self, finding: Finding, platform_enabled: bool):
+        self.__send_pushover_message(finding, platform_enabled)
+
+    def __send_pushover_message(self, finding: Finding, platform_enabled: bool):
+        if self.send_files:
```

Comment:
I think this function could be simplified to the below. Basically, first send the finding details (type of issue, related resource, severity etc), the the other blocks
```
    def __send_pushover_message(self, finding: Finding, platform_enabled: bool):
        # first send finding data
        title, message = self.__get_message_text(finding, platform_enabled)
        investigate_url = finding.get_investigate_uri(self.account_id, self.cluster_name) if platform_enabled else ""
        self.client.send_message(
            title=title, message=message, send_as_html=self.send_as_html, additional_url=investigate_url
        )

        if self.send_files:
            for enrichment in finding.enrichments:
                file_blocks = [block for block in enrichment.blocks if isinstance(block, FileBlock)]
                image_blocks = [block for block in file_blocks if is_image(block.filename)]
                text_blocks = [block for block in file_blocks if block.filename.endswith((".txt", ".log"))]

                if len(image_blocks) > 0:
                    for block in image_blocks:
                        self.client.send_file(
                            file_name=block.filename,
                            contents=block.contents,
                            title=title,
                            message=message,
                            send_as_html=self.send_as_html,
                            additional_url=investigate_url,
                        )

                if len(text_blocks) > 0:
                    for block in text_blocks:
                        message += f"<b>Additional content:</b> \n\n {block.contents.decode()}"

                        self.client.send_message(
                            title=title,
                            message=(message[:1024] if len(message) > 1024 else message),
                            send_as_html=self.send_as_html,
                            additional_url=investigate_url,
                        )

```

Reply:
That looks much nicer and cleaner. Let me try this.